### PR TITLE
Pr build final

### PR DESCRIPTION
**Pull request by leonidcliqz**
Pull request automated builder.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,27 +1,392 @@
...
-CQZ_GOOGLE_API_KEY_CREDENTIAL_ID = 'google-api-key'
-CQZ_MOZILLA_API_KEY_CREDENTIAL_ID = 'mozilla-api-key'
-CQZ_AWS_CREDENTIAL_ID = 'aws-username-and-pass'
-DEBIAN_GPG_KEY_CREDENTIAL_ID = 'debian-gpg-key'
-DEBIAN_GPG_PASS_CREDENTIAL_ID = 'debian-gpg-pass'
-AWS_REGION = 'us-east-1'
 DOCKER_REGISTRY_URL = 'https://141047255820.dkr.ecr.us-east-1.amazonaws.com'
+REPO_URL = 'https://github.com/cliqz-oss/browser-f.git'
+CQZ_BUILD_ID = new Date().format('yyyyMMddHHmmss')
+IMAGE_NAME = 'cliqz/ansible:1101201701'
```
Comment (chrmod) [inline Jenkinsfile]: should be parameterized

File: Jenkinsfile [hunk 2]
```diff
@@ -1,27 +1,392 @@
...
 COMMIT_ID = BUILD_ID
 CQZ_BUILD_DE_LOCALIZATION = ''
-CQZ_GOOGLE_API_KEY_CREDENTIAL_ID = 'google-api-key'
-CQZ_MOZILLA_API_KEY_CREDENTIAL_ID = 'mozilla-api-key'
-CQZ_AWS_CREDENTIAL_ID = 'aws-username-and-pass'
-DEBIAN_GPG_KEY_CREDENTIAL_ID = 'debian-gpg-key'
-DEBIAN_GPG_PASS_CREDENTIAL_ID = 'debian-gpg-pass'
-AWS_REGION = 'us-east-1'
 DOCKER_REGISTRY_URL = 'https://141047255820.dkr.ecr.us-east-1.amazonaws.com'
+REPO_URL = 'https://github.com/cliqz-oss/browser-f.git'
```
Comment (chrmod) [inline Jenkinsfile]: unused?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,27 +1,392 @@
...
+        }
     }
+    // Try to get a jenkins slave. If the slave was created by this action
+    // we need to bootstrap it with ansible
+    def ec2_node = helpers.getEC2Slave("c:/jenkins")
+
+    if (ec2_node.get('created')) {
+        echo "New slave created. Starting provisioning"
+
+        node('docker') {
```
Comment (chrmod) [inline Jenkinsfile]: must be merged with previous `node('docker')` section, ATM a different node may be picked, so sources may not be there

File: Jenkinsfile [hunk 4]
```diff
@@ -1,27 +1,392 @@
...
+            ws() {
+                withCredentials([
+                [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: params.CQZ_AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
```
Comment (chrmod) [inline Jenkinsfile]: please wrap the long lines. 

we can discuss a code style for groovy if you like

File: Jenkinsfile [hunk 5]
```diff
@@ -1,27 +1,392 @@
...
+                    def bootstrap_args = "-u 0 -e aws_access_key=${AWS_ACCESS_KEY_ID} -e aws_secret_key=${AWS_SECRET_ACCESS_KEY} -e instance_name=${ec2_node.get('nodeId')}"
+                    echo "Running with ${bootstrap_args} params"
+                    sh "`aws ecr get-login --region=${params.AWS_REGION}`"
```
Comment (chrmod) [inline Jenkinsfile]: no need to use backticks (`)

File: Jenkinsfile [hunk 6]
```diff
@@ -1,27 +1,392 @@
...
+
+    if (ec2_node.get('created')) {
+        echo "New slave created. Starting provisioning"
+
+        node('docker') {
+            ws() {
+                withCredentials([
+                [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: params.CQZ_AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                    def bootstrap_args = "-u 0 -e aws_access_key=${AWS_ACCESS_KEY_ID} -e aws_secret_key=${AWS_SECRET_ACCESS_KEY} -e instance_name=${ec2_node.get('nodeId')}"
+                    echo "Running with ${bootstrap_args} params"
```
Comment (chrmod) [inline Jenkinsfile]: should be printed by the docker command anyways.

File: Jenkinsfile [hunk 7]
```diff
@@ -1,27 +1,392 @@
...
+    node('chromium_mac_buildserver') {
+        ws('x') {
+            stage('OSX Hypervisor Checkout') {
+                checkout scm
+            }
+                             
+            def LANG_PARAM = ""
+            try {
+                if (env.CQZ_LANG) {
+                    LANG_PARAM = "-lang ${CQZ_LANG}"
```
Comment (chrmod) [inline Jenkinsfile]: that should be done once for all builds (win, mac, lin)

File: Jenkinsfile [hunk 8]
```diff
@@ -1,27 +1,392 @@
...
+                }
+            } catch(e) {}
+                
+            stage('OSX Bootstrap') {
+                sh '/bin/bash -lc "pip install compare-locales"'
```
Comment (chrmod) [inline Jenkinsfile]: there is an alternative syntax for stuff like that:
```
sh '''#!/bin/bash -lc
  pip install compare-locales
  python mozilla-release ....
'''
```

File: Jenkinsfile [hunk 9]
```diff
@@ -1,27 +1,392 @@
...
+                "CQZ_BUILD_ID=$CQZ_BUILD_ID",
+                "CQZ_BUILD_DE_LOCALIZATION=$CQZ_BUILD_DE_LOCALIZATION",
+                "CQZ_RELEASE_CHANNEL=$CQZ_RELEASE_CHANNEL"]) {
+
+                stage('OSX Build') {
+                     withCredentials([
+                         [$class: 'StringBinding', credentialsId: params.CQZ_GOOGLE_API_KEY_CREDENTIAL_ID, variable: 'CQZ_GOOGLE_API_KEY'],
+                         [$class: 'StringBinding', credentialsId: params.CQZ_MOZILLA_API_KEY_CREDENTIAL_ID, variable: 'MOZ_MOZILLA_API_KEY']]) {
+
+                         sh '/bin/bash -lc "./magic_build_and_package.sh --clobber ${LANG_PARAM}"'
```
Comment (chrmod) [inline Jenkinsfile]: would be best if build node be configured so bash is its default shell

File: Jenkinsfile [hunk 10]
```diff
@@ -1,27 +1,392 @@
...
+                    sh '/bin/bash -lc "rm -rf obj/pkg"'
+
+                    withCredentials([
+                        [$class: 'FileBinding', credentialsId: params.MAC_CERT_CREDENTIAL_ID, variable: 'CERT_FILE'],
+                        [$class: 'StringBinding', credentialsId: params.MAC_CERT_PASS_CREDENTIAL_ID, variable: 'CERT_PASS']
+                    ]) {
+                        try {
+                            // create temporary keychain and make it a default one
+                            sh '''#!/bin/bash -l -x
+                                security create-keychain -p cliqz cliqz
```
Comment (chrmod) [inline Jenkinsfile]: side note: we may like to integrate it with `sign_mac.sh`

File: Jenkinsfile (hunk 3 above)
Comment (leonidcliqz) [inline Jenkinsfile]: This section only runs if node should be provisioned, and it doesn't have to checkout the scm again. This is why its in two different sections

**Pull request merged**